patternfly/patternfly-elements
# fix(switch): workaround for double-event firing in ff

## What I did
I noticed that in Firefox (only), when switch is nested inside a `<label>`, then clicking the switch fires the onClick handler twice with two different events.

This PR checks the event for the existence of `event.explicitOriginalTarget` ([ff only](https://developer.mozilla.org/en-US/docs/Web/API/Event/explicitOriginalTarget)), and if it exists, runs some heuristics in order of increasing performance burden to check if this event is the doppelgänger event
Closes #2209

## Notes to reviewers
This patch contains some mildly weird javascript (assignment expressions returning their RHS operand). This is so that performance-sensitive work only gets done when needed inside the workaround.

## Testing instructions

load the demo page on all three browsers and check that clicking on the label and on the control toggles the state. So that means for each browser checking that both the first instance of switch, the unlabeled instance of switch, and the nested-in-label instance of switch all work as expected.

Note that in safari, clicking the label anywhere doesn't toggle the state, but let's leave that for a subsequent issue.
